pkgs/development/compilers/ghc/8.6.5-binary.nix

```
@@ -12,7 +12,8 @@ assert stdenv.targetPlatform == stdenv.hostPlatform;
 let
   useLLVM = !stdenv.targetPlatform.isx86;
 
-  useNcurses6 = stdenv.hostPlatform.system == "x86_64-linux";
+  useNcurses6 = stdenv.hostPlatform.system == "x86_64-linux"
+                || stdenv.hostPlatform.isPower64;
```

Otherwise there's an ambiguity w.r.t. endianness, is there not?

```suggestion
                || stdenv.hostPlatform.system == "powerpc64le-linx";
```

---

Well, that's intentional... I can't imagine that endianness affects whether or not `ncurses6` should be used.  Frankly I'm not sure under what conditions that variable should be set or why (if you know please tell me and I will add a comment) but it seems to have something to do with 64-bit systems, which is why I set it to `true` for 64-bit powerpc.  Also, it does in fact work correctly... I have `nix-tree` now!  Hooray!

In general we should be moving away from using `builtins.system`; it's sort of `nix`'s ad-hoc platform taxonomy that doesn't represent things like abi, libc.

Also I think you have a typo there with `linx`.  That's another reason to use the predicates from `inspect.nix` rather than string comparison.

---

Yes, the library used does not depend on on the architecture, but that doesn't mean we can freely choose it! It depends on what version of the library the bindist was linked against before it was released, so it needs to be checked *per bindist* and an assumption based on CPU family won't give you a correct result. (I checked and Fedora 29 which this bindist was built using used to ship ncurses 6.)

---

Ah, okay.  Updated to account for this.

We should probably not have a default value here then, and should `throw` if the platform is not one of those for which somebody has manually checked this.

---

Probably, yeah. The newer bindists >= 8.10.2 have a better logic for this, but 8.6.5 was never touched, as it's simple and we thought it wouldn't stick around for too long.